### PR TITLE
[Test] increase e2e test confidence

### LINKED ISSUE #2124: E2E tests
Add the following E2E tests:

## User page
- [ ] Ellipses are shown for very long links

## URL creation
- [ ] It should prevent uploading when a [malicious file](https://www.webroot.com/blog/2018/09/05/eicar-common-false-positive-world/) is submitted
 
## URL Searching and Download (?)
- [ ] Clicking on the download button downloads all links currently shown in the links table as a `.csv` file.

## User page - Sort & Filter
- [ ] Links should be sorted by their number of clicks in descending order when enabling sort by `Most number of visits` and clicking apply

### PR DESCRIPTION
**Pull request by oversparkling**
## Problem

Closes #2124 

**Features**:
- Increases code confidence of our repository, especially our virus scanner on files

**Noticable changes from this PR**
- Local E2E tests may fail locally if there is a `urls.csv` lurking in downloads folder
- Changes made to `docker-compose` will require a `.env` file to be present in local directory (KIV on this approach atm)

## Tests

_What tests should be run to confirm functionality?_
Contribution to E2E Tests

### CODE REVIEW

File: test/end-to-end/util/LinkCreationProcedure.ts [hunk 1]
```diff
@@ -31,13 +31,15 @@ const fetchLink = async (url, numberOfFetches) => {
...
   }
 
-  Promise.all(new Array(numberOfFetches).fill(get(url))).then(
-    (values: boolean[]) => {
-      console.log(
-        `Url: ${url} was fetched ${values.filter(Boolean).length} times`,
-      )
-    },
-  )
+  const fetchArray: Promise<boolean>[] = []
```
Comment (oversparkling) [inline test/end-to-end/util/LinkCreationProcedure.ts:34]: Context behind the bug: The way this array was populated only made the `fetch` call once despite the `numberOfFetches` specified. I suspect that the underlying implementation of this method creates a single instance of `get(url)` and populates the array with the same object references, hence only needing a single promise to be resolved/

File: package.json [hunk 2]
```diff
@@ -59,6 +59,7 @@
     "copy-to-clipboard": "^3.3.1",
     "core-js": "^3.16.3",
     "cross-fetch": "^3.1.5",
+    "csv-parse": "^5.3.6",
```
Comment (gweiying) [inline package.json:62]: suggest installing this only as a devDependency instead of a dependency since we're only using it for our e2e tests

**Comment by gweiying** [comment]
Noting that above tests addresses these test cases:

User page
- [ ] Ellipses are shown for very long links

URL creation
- [ ] It should prevent uploading when a [malicious file](https://www.webroot.com/blog/2018/09/05/eicar-common-false-positive-world/) is submitted
 
URL Searching and Download (?)
- [ ] Clicking on the download button downloads all links currently shown in the links table as a `.csv` file.

User page - Sort & Filter
- [ ] Links should be sorted by their number of clicks in descending order when enabling sort by `Most number of visits` and clicking apply

**Comment by oversparkling** [comment]
Referencing [this](https://stackoverflow.com/a/74069689), we removed assignment of empty variables in `docker-compose` so that it defaults to `.env` file that is created during CI.

**Review by gweiying — APPROVED** [review]
lgtm!

**Pull request merged**